.github/actions/build_squashfs_image/action.yml

```diff
@@ -0,0 +1,80 @@
+name: "Build SquashFS Image"
+description: "Builds a SquashFS image using Docker"
+inputs:
+  platform:
+    description: "The platform to build for"
+    required: true
+  platform_suffix:
+    description: "The platform suffix"
+    required: true
+  base_image:
+    description: "The base image to use"
+    required: true
+  whl:
+    description: "The OTA client wheel file"
+    required: true
+  version:
+    description: "The OTA client version"
+    required: true
+  output_squashfs:
+    description: "The output SquashFS file"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: download artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: artifacts-python-packages
+        path: dist
+
+    - name: set temporary variables
+      id: set-vars
+      env:
+        PLATFORM_SUFFIX: ${{ inputs.platform_suffix }}
+        VERSION: ${{ inputs.version }}
+      shell: bash
+      run: |
+        echo "DOCKER_IMAGE=otaclient-${PLATFORM_SUFFIX}:${VERSION}" >> $GITHUB_ENV
+        echo "DOCKER_CONTAINER=otaclient-${PLATFORM_SUFFIX}_v${VERSION}" >> $GITHUB_ENV
+
+    - name: build docker image
+      env:
+        DOCKER_IMAGE: ${{ env.DOCKER_IMAGE }}
+        PLATFORM: ${{ inputs.platform }}
+        PLATFORM_SUFFIX: ${{ inputs.platform_suffix }}
+        BASE_IMAGE: ${{ inputs.base_image }}
+        WHL: ${{ inputs.whl }}
+        VERSION: ${{ inputs.version }}
+      shell: bash
+      run: |
+        cp dist/${WHL} docker/app_img/
+        pushd docker/app_img
+        docker buildx create --use
+        docker buildx build --platform ${PLATFORM} \
+          --build-arg=UBUNTU_BASE=${BASE_IMAGE} \
+          --build-arg=OTACLIENT_VERSION=${VERSION} \
+          --build-arg=OTACLIENT_WHL=${WHL} \
+          -t ${DOCKER_IMAGE} \
+          --output type=docker \
+          .
+        popd
```

Comment:
use `docker buildx` for ARM.
In future, once ARM runners are available, will be better to migrate to use the runner rather than qemu.